lib/bridge/MatrixHandler.js

```diff
@@ -1261,7 +1263,52 @@ MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
         return;
     }
 
+    /**
+     * Cache events in here so we can refer to them for replies.
+     * We deliberately avoid fetching events from the homeserver due to speed concerns.
+     */
+    this._eventCache.set(event.event_id, { body: event.content.body, sender: event.sender });
```

Comment:
There's no protection from massive events landing in the cache.

Reply:
True, we should trim this to how much of the body we actually need.